simulation/revolve2/simulation/scene/geometry/textures/_texture_reference.py

```diff
@@ -0,0 +1,31 @@
+from abc import ABC
+from dataclasses import dataclass, field
+
+
+@dataclass(frozen=True, kw_only=True)
+class TextureReference(ABC):
+    """Reference texture information for the simulators."""
+
+    builtin: str | None = field(default=None)
+    """Allows to reference builtin textures of the targeted simulator."""
+    file: str | None = field(default=None)
+    """The source file of the specified texture, if custom."""
+    content_type: str | None = field(default=None)
+    """The type of the source file. The supported types depend on the simulator used. Please check the simulators documentation for supported filetypes."""
+    gridlayout: str | None = field(default=None)
+    """Specify the layout of the texture to be mapped onto an object. This again depends on how the simulator implements grid layouts. Therefore refer to the simulators documentation."""
```

Comment:
```suggestion
    """Specify the layout of the texture to be mapped onto an object. This depends on the simulator used. Please check the simulators documentation for supported grid layouts."""
```

Best to have docstrings not be dependent on each other